### PR TITLE
Add FusedSDPA QKV slice support for APC

### PR DESCRIPTION
**Pull request by ikurtchen**
The function can be controlled by following env:

- VLLM_HPU_FSDPA_SLICE_SEQ_LEN_THLD (int):
  -     QKV slice threshold, default is 0 which means disabled. 
- VLLM_HPU_FSDPA_SLICE_CHUNK_SIZE (int):
  -     QKV slice size, default 4096.
- VLLM_HPU_FSDPA_SLICE_IMPL (str):
  -     slice_qkv (default): split prefix & causal, and do QKV slice on both parts.
  -     slice_causal: split prefix & causal, and do QKV slice on causal part only.
  -     split_kv: split prefix & causal, no QKV slice.
- VLLM_HPU_FSDPA_SLICE_CAUSAL (boolean):
  -     Also do QKV slice in non APC case when query len greater than threshold, default: False.

Accuracy verified with Qwen3-32B. Perf improvement is ~40% with 10K prefix and 12K query.

### CODE REVIEW

File: vllm_hpu_extension/ops.py [hunk 1]
```diff
@@ -394,7 +460,7 @@ def prompt_attention(
         impl: str,
         **args,
 ) -> torch.Tensor:
-    _get_context(args)
+    _get_context(impl, args)
```
Comment (czhu15) [inline vllm_hpu_extension/ops.py]: can we put impl into args? so we can keep _get_context() interface unchanged, and no impact to other function calls this _get_context.

File: vllm_hpu_extension/ops.py [hunk 2]
```diff
@@ -1381,3 +1470,561 @@ def scaled_fp8_quant(
...
+        kv_len = key.size(-2)
+
+        assert query_len == kv_len, f"For causal SDPA, {query_len=} should be equal to {kv_len=}"
+
+        gqa = is_gqa(query, key)
+        if gqa:
+            query, key, value, attn_mask = gqa_input_reshape_fwd(
+                query, key, value, attn_mask)
+
+        if query_len % 1024 != 0:
```
Comment (czhu15) [inline vllm_hpu_extension/ops.py:1826]: can you add a comment why need special handling for "if query_len % 1024 != 0"

**Review by czhu15 — APPROVED** [review]
LTMG except one minor comment.
+ @yangulei for review.

File: vllm_hpu_extension/ops.py (hunk 2 above)
Comment (ikurtchen) [reply vllm_hpu_extension/ops.py:1826]: It's from @yangulei , we cannot use causal to get correct m/linv when query len which cannot div 1024. Similar to:

https://github.com/intel/neural-compressor/commit/86c548d977eef540375f5bce0ee94a79442ad434#diff-8ea59b49d024c85c51ac4a5b6937ac9f253a868b308cfa98af0a8016dd0abdaeR465

File: vllm_hpu_extension/ops.py [hunk 3]
```diff
@@ -1381,3 +1470,561 @@ def scaled_fp8_quant(
...
+            is_causal = False
+            if attn_mask is None:
+                bs = query.size(0)
+                mask = (1 - torch.tril(
+                    torch.ones(bs,
+                               1,
+                               1,
+                               query_len,
+                               query_len,
```
Comment (yangulei) [inline vllm_hpu_extension/ops.py]: Did you test using the GQA and non-GQA models? The mask should have different dimensions.

Comment (ikurtchen) [reply vllm_hpu_extension/ops.py]: Yes, it needs to handle MHA case. Fixed and verified with LLama2-7B.

**Comment by czhu15** [comment]
can you also update PR description?
"
QKV slice threshold, default 8192, set 0 to disable. 
"
Now default is disabled, i.e. 0.

**Comment by ikurtchen** [comment]
> can you also update PR description? " QKV slice threshold, default 8192, set 0 to disable. " Now default is disabled, i.e. 0.

Done, updated PR desc.

**Review by yangulei — APPROVED** [review]
LGTM
